### PR TITLE
Add unit test case for sharding-orchestration-center-apollo module #4451

### LINKED ISSUE #4451: Add unit test case for sharding-orchestration-center-apollo module
Detailed task in issue   #4439

### PR DESCRIPTION
**Pull request by gongsiran**
Fixes #4451.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 1]
```diff
@@ -82,4 +91,65 @@ public void onChange(final DataChangedEvent dataChangedEvent) {
...
         assertThat(changeEvent.getChangedType(), is(DataChangedEvent.ChangedType.UPDATED));
     }
+
+    @Test
+    @SneakyThrows
+    public void assertGetWithNonExistentKey() {
+        when(openApiWrapper.getValue(eq("test.nonExistentKey"))).thenReturn(null);
+        assertNull(configCenterRepository.get("/test/nonExistentKey"));
+    }
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: Keep four blank characters.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 2]
```diff
@@ -82,4 +91,65 @@ public void onChange(final DataChangedEvent dataChangedEvent) {
...
+                future.set(dataChangedEvent);
+            }
+        });
+        embeddedApollo.addOrModifyProperty("orchestration", "test.children.1", "newValue1");
+        DataChangedEvent changeEvent = future.get(5, TimeUnit.SECONDS);
+        assertThat(changeEvent.getKey(), is("/test/children/1"));
+        assertThat(changeEvent.getValue(), is("newValue1"));
+        assertThat(changeEvent.getChangedType(), is(DataChangedEvent.ChangedType.UPDATED));
+    }
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: Keep four blank characters. Please check every method.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 3]
```diff
@@ -82,4 +91,65 @@ public void onChange(final DataChangedEvent dataChangedEvent) {
...
+            @Override
+            public void onChange(final DataChangedEvent dataChangedEvent) {
+                future.set(dataChangedEvent);
+            }
+        });
+        embeddedApollo.deleteProperty("orchestration", "test.children.1");
+        DataChangedEvent changeEvent = future.get(5, TimeUnit.SECONDS);
+        assertThat(changeEvent.getKey(), is("/test/children/1"));
+        assertNull(changeEvent.getValue());
+        assertThat(changeEvent.getChangedType(), is(DataChangedEvent.ChangedType.DELETED));
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: now we support delete?

Comment (gongsiran) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: we not support   delete ，but can  add dataChangedEventListener

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 4]
```diff
@@ -67,12 +78,62 @@ public void assertGet() {
...
-        configCenterRepository.watch("/test/children/1", future::set);
+        configCenterRepository.watch("/test/children/1", new DataChangedEventListener() {
+    
+            @Override
+            public void onChange(final DataChangedEvent dataChangedEvent) {
+                future.set(dataChangedEvent);
+            }
+        });
+        embeddedApollo.deleteProperty("orchestration", "test.children.1");
         embeddedApollo.addOrModifyProperty("orchestration", "test.children.1", "value3");
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: Why delete the property and then add it right now?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 5]
```diff
@@ -67,12 +78,62 @@ public void assertGet() {
...
-        assertThat(changeEvent.getValue(), is("value3"));
+        assertNull(changeEvent.getValue());
+        assertThat(changeEvent.getChangedType(), is(DataChangedEvent.ChangedType.DELETED));
+        assertNull(configCenterRepository.get("/test/children/1"));
+    }
+    
+    @Test
+    @SneakyThrows
+    public void assertWatchAddChangedType() {
+        assertWatchUpdateChangedType("/test/children/newKey", "newVaule");
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: It's not a good idea for using `assertWatchUpdateChangedType ` method  at `assertWatchAddChangedType ` method, maybe we should rename these methods and make them more meaningful.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 6]
```diff
@@ -67,12 +78,62 @@ public void assertGet() {
...
+        assertThat(configCenterRepository.get("/test/children/1"), is("newValue1"));
+    }
+    
+    @Test
+    @SneakyThrows
+    public void assertWatchDeletedChangedType() {
         final SettableFuture<DataChangedEvent> future = SettableFuture.create();
-        configCenterRepository.watch("/test/children/1", future::set);
+        configCenterRepository.watch("/test/children/1", new DataChangedEventListener() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: Keep this line's blank character with `@Override`.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java [hunk 7]
```diff
@@ -67,12 +78,62 @@ public void assertGet() {
...
+    @SneakyThrows
+    public void assertWatchAddChangedType() {
+        assertWatchUpdateChangedType("/test/children/newKey", "newVaule");
+    }
+    
+    @SneakyThrows
+    private void assertWatchUpdateChangedType(final String key, final String newVaule) {
+        final SettableFuture<DataChangedEvent> future = SettableFuture.create();
+        configCenterRepository.watch(key, new DataChangedEventListener() {
+
```
Comment (dongzl) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstanceTest.java]: Keep this line's blank character with `@Override`.

**Pull request merged**